articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+
+    oc new-app rails-postgresql-example
+
+    to build a new example application in Ruby. Or use kubectl to deploy a simple Kubernetes application:
+
+    kubectl create deployment hello-node --image=k8s.gcr.io/serve_hostname
+    ```
+
+1. Execute the following command to add the view role to the default service account. This role is needed so the application can discover other pods and form a cluster with them:
+
+    ```bash
+    $ oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -n $(oc project -q)
+    clusterrole.rbac.authorization.k8s.io/view added: "system:serviceaccount:eap-demo:default"
+    ```
+
+## Prepare the application
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from your application to the database server available on your environment.
+
+When you are targeting OpenShift, you might want to trim the capabilities of your JBoss EAP server, for example, to reduce the security exposure of the provisioned server and reduce the overall footprint. You could also want to include some Microprofile specs to make your application more suitable for running on an OpenShift environment. When using JBoss EAP, one way to accomplish this is by packaging your application and your server in a single deployment unit known as Bootable JAR. Let's do that by adding the required changes to our demo application.
+
+Navigate to your demo application local repository and change the branch to `bootable-jar`:
+
+```bash
+todo-list (main) $ git checkout bootable-jar
+Switched to branch 'bootable-jar'
+todo-list (bootable-jar) $
+```
+
+Let's do a quick review about what we have changed:
+
+1. We have added the `wildfly-jar-maven` plugin to provision the server and the application in a single executable JAR file. The OpenShift deployment unit will be now our server with our application.
+1. On the maven plugin, we have specified a set of Galleon layers. This configuration allows us to trim the server capabilities to only what we need.
+1. Our application uses JSF with Ajax requests, which means there will be information stored in the HTTP Session. We don't want to lose such information if a pod is removed. We could save this information on the client and send it back on each request. However, there are cases where you may not distribute certain information to the clients. For this demo, we have chosen to replicate the session across all pod replicas. To do it, we have added `<distributable />` on the `web.xml` that together with the server clustering capabilities will make the HTTP Session distributable across all pods.
```

Comment:
s/JSF/Jakarta Faces/

Reply:
s/However, there are cases where you may not distribute certain information to the clients./However, there are cases where you may decide not to distribute certain information to the clients./